raven/versioning.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+
+__all__ = ('fetch_git_sha', 'fetch_package_version')
+
+import os.path
+import pkg_resources
+
+from .exceptions import InvalidGitRepository
+
+
+def fetch_git_sha(path, head='master'):
+    """
+    >>> fetch_git_sha(os.path.dirname(__file__))
+    """
+    revision_file = os.path.join(path, '.git', 'refs', 'heads', 'master')
```

Comment:
The problem with .git/HEAD is/was probably that it needs to get parsed/followed when it starts with `ref:` etc.

I found the following to be useful for me, but it requires `git` to be installed:

    subprocess.check_output(['git', 'describe', '--always', '--dirty']).rstrip()